commit abb22103aba817d1a0916ed758ecf38baac7083e
Author: Susan Franklin <sfranklin>
Date:   Wed Apr 23 21:42:58 2008 +0000

    Bug 228481 - [ui] ProfileModificationWizards should report and allow continuation of INFO and WARNING plans

6	6	bundles/org.eclipse.equinox.p2.ui/src/org/eclipse/equinox/internal/p2/ui/dialogs/ProfileModificationWizardPage.java
diff --git a/bundles/org.eclipse.equinox.p2.ui/src/org/eclipse/equinox/internal/p2/ui/dialogs/ProfileModificationWizardPage.java b/bundles/org.eclipse.equinox.p2.ui/src/org/eclipse/equinox/internal/p2/ui/dialogs/ProfileModificationWizardPage.java
index c2290cd..d51913b 100644
--- a/bundles/org.eclipse.equinox.p2.ui/src/org/eclipse/equinox/internal/p2/ui/dialogs/ProfileModificationWizardPage.java
+++ b/bundles/org.eclipse.equinox.p2.ui/src/org/eclipse/equinox/internal/p2/ui/dialogs/ProfileModificationWizardPage.java
@@ -208,7 +208,7 @@ public abstract class ProfileModificationWizardPage extends WizardPage {
 			final Object[] selections = getCheckedElements();
 			if (selections.length == 0) {
 				currentPlan = null;
-				currentStatus = new Status(IStatus.WARNING, ProvUIActivator.PLUGIN_ID, ProvUIMessages.ProfileModificationWizardPage_NothingSelected);
+				currentStatus = new Status(IStatus.ERROR, ProvUIActivator.PLUGIN_ID, ProvUIMessages.ProfileModificationWizardPage_NothingSelected);
 			} else
 				getContainer().run(true, true, new IRunnableWithProgress() {
 					public void run(IProgressMonitor monitor) {
@@ -256,18 +256,18 @@ public abstract class ProfileModificationWizardPage extends WizardPage {
 
 	void updateStatus() {
 		int messageType = IMessageProvider.NONE;
+		boolean pageComplete = true;
 		if (currentStatus != null && !currentStatus.isOK()) {
 			messageType = IMessageProvider.INFORMATION;
 			int severity = currentStatus.getSeverity();
-			if (severity == IStatus.ERROR)
+			if (severity == IStatus.ERROR) {
 				messageType = IMessageProvider.ERROR;
-			else if (severity == IStatus.WARNING)
+				pageComplete = false;
+			} else if (severity == IStatus.WARNING)
 				messageType = IMessageProvider.WARNING;
-			setPageComplete(false);
 			ProvUI.reportStatus(currentStatus, StatusManager.LOG);
-		} else {
-			setPageComplete(true);
 		}
+		setPageComplete(pageComplete);
 		setMessage(getMessageText(), messageType);
 		detailsArea.setText(getDetailText());
 	}
commit bbb8d7976ff2443524a174666b08cc33a471847b
Author: Susan Franklin <sfranklin>
Date:   Wed Apr 23 21:53:57 2008 +0000

    Bug 228481 - [ui] ProfileModificationWizards should report and allow continuation of INFO and WARNING plans

8	4	bundles/org.eclipse.equinox.p2.ui.sdk/src/org/eclipse/equinox/internal/p2/ui/sdk/ProvSDKUIActivator.java
1	1	bundles/org.eclipse.equinox.p2.ui.sdk/src/org/eclipse/equinox/internal/p2/ui/sdk/prefs/PreferenceConstants.java
1	1	bundles/org.eclipse.equinox.p2.ui.sdk/src/org/eclipse/equinox/internal/p2/ui/sdk/prefs/PreferenceInitializer.java
5	5	bundles/org.eclipse.equinox.p2.ui.sdk/src/org/eclipse/equinox/internal/p2/ui/sdk/prefs/ProvisioningPreferencePage.java
8	6	bundles/org.eclipse.equinox.p2.ui.sdk/src/org/eclipse/equinox/internal/p2/ui/sdk/updates/AutomaticUpdater.java
diff --git a/bundles/org.eclipse.equinox.p2.ui.sdk/src/org/eclipse/equinox/internal/p2/ui/sdk/ProvSDKUIActivator.java b/bundles/org.eclipse.equinox.p2.ui.sdk/src/org/eclipse/equinox/internal/p2/ui/sdk/ProvSDKUIActivator.java
index 2ad947e..6f48c9f 100644
--- a/bundles/org.eclipse.equinox.p2.ui.sdk/src/org/eclipse/equinox/internal/p2/ui/sdk/ProvSDKUIActivator.java
+++ b/bundles/org.eclipse.equinox.p2.ui.sdk/src/org/eclipse/equinox/internal/p2/ui/sdk/ProvSDKUIActivator.java
@@ -217,8 +217,6 @@ public class ProvSDKUIActivator extends AbstractUIPlugin {
 							});
 						return false;
 					}
-					if (plan.getStatus().isOK())
-						return true;
 
 					// Special case those statuses where we would never want to open a wizard
 					if (plan.getStatus().getCode() == IStatusCodes.NOTHING_TO_UPDATE) {
@@ -226,7 +224,12 @@ public class ProvSDKUIActivator extends AbstractUIPlugin {
 						return false;
 					}
 
-					String openPlan = getPreferenceStore().getString(PreferenceConstants.PREF_OPEN_WIZARD_ON_NONOK_PLAN);
+					// Allow the wizard to open if there is no error
+					if (plan.getStatus().getSeverity() != IStatus.ERROR)
+						return true;
+
+					// There is an error.  Check the preference to see whether to continue.
+					String openPlan = getPreferenceStore().getString(PreferenceConstants.PREF_OPEN_WIZARD_ON_ERROR_PLAN);
 					if (MessageDialogWithToggle.ALWAYS.equals(openPlan)) {
 						return true;
 					}
@@ -234,7 +237,8 @@ public class ProvSDKUIActivator extends AbstractUIPlugin {
 						ProvUI.reportStatus(plan.getStatus(), StatusManager.SHOW | StatusManager.LOG);
 						return false;
 					}
-					MessageDialogWithToggle dialog = MessageDialogWithToggle.openYesNoCancelQuestion(shell, ProvSDKMessages.ProvSDKUIActivator_Question, ProvSDKMessages.ProvSDKUIActivator_OpenWizardAnyway, null, false, getPreferenceStore(), PreferenceConstants.PREF_OPEN_WIZARD_ON_NONOK_PLAN);
+					MessageDialogWithToggle dialog = MessageDialogWithToggle.openYesNoCancelQuestion(shell, ProvSDKMessages.ProvSDKUIActivator_Question, ProvSDKMessages.ProvSDKUIActivator_OpenWizardAnyway, null, false, getPreferenceStore(), PreferenceConstants.PREF_OPEN_WIZARD_ON_ERROR_PLAN);
+
 					// Any answer but yes will stop the performance of the plan, but NO is interpreted to mean, show me the error.
 					if (dialog.getReturnCode() == IDialogConstants.NO_ID)
 						ProvUI.reportStatus(plan.getStatus(), StatusManager.SHOW | StatusManager.LOG);
diff --git a/bundles/org.eclipse.equinox.p2.ui.sdk/src/org/eclipse/equinox/internal/p2/ui/sdk/prefs/PreferenceConstants.java b/bundles/org.eclipse.equinox.p2.ui.sdk/src/org/eclipse/equinox/internal/p2/ui/sdk/prefs/PreferenceConstants.java
index f070d55..b396d4d 100644
--- a/bundles/org.eclipse.equinox.p2.ui.sdk/src/org/eclipse/equinox/internal/p2/ui/sdk/prefs/PreferenceConstants.java
+++ b/bundles/org.eclipse.equinox.p2.ui.sdk/src/org/eclipse/equinox/internal/p2/ui/sdk/prefs/PreferenceConstants.java
@@ -24,7 +24,7 @@ public class PreferenceConstants {
 	public static final String PREF_REMIND_SCHEDULE = "remindOnSchedule"; // value is true or false //$NON-NLS-1$
 	public static final String PREF_REMIND_ELAPSED = "remindElapsedTime";//$NON-NLS-1$
 	public static final String PREF_SHOW_LATEST_VERSION = "showLatestVersion"; //$NON-NLS-1$
-	public static final String PREF_OPEN_WIZARD_ON_NONOK_PLAN = "allowNonOKPlan"; //$NON-NLS-1$
+	public static final String PREF_OPEN_WIZARD_ON_ERROR_PLAN = "allowNonOKPlan"; //$NON-NLS-1$
 	public static final String PREF_GENERATE_ARCHIVEREPOFOLDER = "generateReposForArchive"; //$NON-NLS-1$
 	public static final String PREF_AUTO_INSTALL_BUNDLES = "autoInstallBundles"; //$NON-NLS-1$
 }
diff --git a/bundles/org.eclipse.equinox.p2.ui.sdk/src/org/eclipse/equinox/internal/p2/ui/sdk/prefs/PreferenceInitializer.java b/bundles/org.eclipse.equinox.p2.ui.sdk/src/org/eclipse/equinox/internal/p2/ui/sdk/prefs/PreferenceInitializer.java
index 4d7b5d4..f7007dd 100644
--- a/bundles/org.eclipse.equinox.p2.ui.sdk/src/org/eclipse/equinox/internal/p2/ui/sdk/prefs/PreferenceInitializer.java
+++ b/bundles/org.eclipse.equinox.p2.ui.sdk/src/org/eclipse/equinox/internal/p2/ui/sdk/prefs/PreferenceInitializer.java
@@ -32,7 +32,7 @@ public class PreferenceInitializer extends AbstractPreferenceInitializer {
 		node.putBoolean(PreferenceConstants.PREF_REMIND_SCHEDULE, false);
 		node.putBoolean(PreferenceConstants.PREF_SHOW_LATEST_VERSION, true);
 		node.put(PreferenceConstants.PREF_REMIND_ELAPSED, ProvSDKMessages.AutomaticUpdateScheduler_30Minutes);
-		node.put(PreferenceConstants.PREF_OPEN_WIZARD_ON_NONOK_PLAN, MessageDialogWithToggle.PROMPT);
+		node.put(PreferenceConstants.PREF_OPEN_WIZARD_ON_ERROR_PLAN, MessageDialogWithToggle.PROMPT);
 		node.put(PreferenceConstants.PREF_GENERATE_ARCHIVEREPOFOLDER, MessageDialogWithToggle.PROMPT);
 		node.put(PreferenceConstants.PREF_AUTO_INSTALL_BUNDLES, MessageDialogWithToggle.PROMPT);
 	}
diff --git a/bundles/org.eclipse.equinox.p2.ui.sdk/src/org/eclipse/equinox/internal/p2/ui/sdk/prefs/ProvisioningPreferencePage.java b/bundles/org.eclipse.equinox.p2.ui.sdk/src/org/eclipse/equinox/internal/p2/ui/sdk/prefs/ProvisioningPreferencePage.java
index 770b8d9..f570a82 100644
--- a/bundles/org.eclipse.equinox.p2.ui.sdk/src/org/eclipse/equinox/internal/p2/ui/sdk/prefs/ProvisioningPreferencePage.java
+++ b/bundles/org.eclipse.equinox.p2.ui.sdk/src/org/eclipse/equinox/internal/p2/ui/sdk/prefs/ProvisioningPreferencePage.java
@@ -100,7 +100,7 @@ public class ProvisioningPreferencePage extends PreferencePage implements IWorkb
 		Preferences pref = ProvSDKUIActivator.getDefault().getPluginPreferences();
 		showLatestRadio.setSelection(pref.getBoolean(PreferenceConstants.PREF_SHOW_LATEST_VERSION));
 		showAllRadio.setSelection(!pref.getBoolean(PreferenceConstants.PREF_SHOW_LATEST_VERSION));
-		String openWizard = pref.getString(PreferenceConstants.PREF_OPEN_WIZARD_ON_NONOK_PLAN);
+		String openWizard = pref.getString(PreferenceConstants.PREF_OPEN_WIZARD_ON_ERROR_PLAN);
 		alwaysShowFailedPlan.setSelection(openWizard.equals(MessageDialogWithToggle.ALWAYS));
 		neverShowFailedPlan.setSelection(openWizard.equals(MessageDialogWithToggle.NEVER));
 		promptOnFailedPlan.setSelection(openWizard.equals(MessageDialogWithToggle.PROMPT));
@@ -111,7 +111,7 @@ public class ProvisioningPreferencePage extends PreferencePage implements IWorkb
 		Preferences pref = ProvSDKUIActivator.getDefault().getPluginPreferences();
 		showLatestRadio.setSelection(pref.getDefaultBoolean(PreferenceConstants.PREF_SHOW_LATEST_VERSION));
 		showAllRadio.setSelection(!pref.getDefaultBoolean(PreferenceConstants.PREF_SHOW_LATEST_VERSION));
-		String openWizard = pref.getDefaultString(PreferenceConstants.PREF_OPEN_WIZARD_ON_NONOK_PLAN);
+		String openWizard = pref.getDefaultString(PreferenceConstants.PREF_OPEN_WIZARD_ON_ERROR_PLAN);
 		alwaysShowFailedPlan.setSelection(openWizard.equals(MessageDialogWithToggle.ALWAYS));
 		neverShowFailedPlan.setSelection(openWizard.equals(MessageDialogWithToggle.NEVER));
 		promptOnFailedPlan.setSelection(openWizard.equals(MessageDialogWithToggle.PROMPT));
@@ -121,11 +121,11 @@ public class ProvisioningPreferencePage extends PreferencePage implements IWorkb
 		Preferences pref = ProvSDKUIActivator.getDefault().getPluginPreferences();
 		pref.setValue(PreferenceConstants.PREF_SHOW_LATEST_VERSION, showLatestRadio.getSelection());
 		if (alwaysShowFailedPlan.getSelection())
-			pref.setValue(PreferenceConstants.PREF_OPEN_WIZARD_ON_NONOK_PLAN, MessageDialogWithToggle.ALWAYS);
+			pref.setValue(PreferenceConstants.PREF_OPEN_WIZARD_ON_ERROR_PLAN, MessageDialogWithToggle.ALWAYS);
 		else if (neverShowFailedPlan.getSelection())
-			pref.setValue(PreferenceConstants.PREF_OPEN_WIZARD_ON_NONOK_PLAN, MessageDialogWithToggle.NEVER);
+			pref.setValue(PreferenceConstants.PREF_OPEN_WIZARD_ON_ERROR_PLAN, MessageDialogWithToggle.NEVER);
 		else
-			pref.setValue(PreferenceConstants.PREF_OPEN_WIZARD_ON_NONOK_PLAN, MessageDialogWithToggle.PROMPT);
+			pref.setValue(PreferenceConstants.PREF_OPEN_WIZARD_ON_ERROR_PLAN, MessageDialogWithToggle.PROMPT);
 
 		ProvSDKUIActivator.getDefault().savePluginPreferences();
 		return true;
diff --git a/bundles/org.eclipse.equinox.p2.ui.sdk/src/org/eclipse/equinox/internal/p2/ui/sdk/updates/AutomaticUpdater.java b/bundles/org.eclipse.equinox.p2.ui.sdk/src/org/eclipse/equinox/internal/p2/ui/sdk/updates/AutomaticUpdater.java
index a708340..20f3f28 100644
--- a/bundles/org.eclipse.equinox.p2.ui.sdk/src/org/eclipse/equinox/internal/p2/ui/sdk/updates/AutomaticUpdater.java
+++ b/bundles/org.eclipse.equinox.p2.ui.sdk/src/org/eclipse/equinox/internal/p2/ui/sdk/updates/AutomaticUpdater.java
@@ -78,20 +78,22 @@ public class AutomaticUpdater implements IUpdateListener {
 					if (alreadyValidated)
 						return true;
 					// In all other cases we return false, because the clicking the popup will actually run the action.
-					String openPlan = prefs.getString(PreferenceConstants.PREF_OPEN_WIZARD_ON_NONOK_PLAN);
+					// We are just checking prefs to determine whether to to show the popup or not.
+					String openPlan = prefs.getString(PreferenceConstants.PREF_OPEN_WIZARD_ON_ERROR_PLAN);
 					if (plan != null) {
-						if (plan.getStatus().isOK() || !(MessageDialogWithToggle.NEVER.equals(openPlan))) {
+						boolean noError = plan.getStatus().getSeverity() != IStatus.ERROR;
+						if (noError || !(MessageDialogWithToggle.NEVER.equals(openPlan))) {
 							// Show the affordance if user prefers always opening a currentPlan or being prompted
 							// In this context, the affordance is the prompt.
 							if (updateAffordance == null)
 								createUpdateAffordance();
-							setUpdateAffordanceState(plan.getStatus().isOK());
-							// If the user always wants to open invalid plans, or status is OK then go ahead and show
+							setUpdateAffordanceState(noError);
+							// If the user always wants to open invalid plans, or there is no error then go ahead and show
 							// the popup.
-							if (plan.getStatus().isOK() || MessageDialogWithToggle.ALWAYS.equals(openPlan) && popup == null)
+							if ((noError || MessageDialogWithToggle.ALWAYS.equals(openPlan)) && popup == null)
 								createUpdatePopup();
 						} else {
-							// The pref is NEVER, the user doesn't want to know about it
+							// There is an error and the pref is NEVER, the user doesn't want to know about it
 							ProvUI.reportStatus(plan.getStatus(), StatusManager.LOG);
 						}
 					}
commit 6d3dcd1e5ea399198cde6157d74a0ba95e4f119a
Author: Susan Franklin <sfranklin>
Date:   Tue Apr 29 20:18:40 2008 +0000

    Bug 228481 - [ui] ProfileModificationWizards should report and allow continuation of INFO and WARNING plans

1	1	bundles/org.eclipse.equinox.p2.ui/src/org/eclipse/equinox/internal/p2/ui/dialogs/ProfileModificationWizardPage.java
diff --git a/bundles/org.eclipse.equinox.p2.ui/src/org/eclipse/equinox/internal/p2/ui/dialogs/ProfileModificationWizardPage.java b/bundles/org.eclipse.equinox.p2.ui/src/org/eclipse/equinox/internal/p2/ui/dialogs/ProfileModificationWizardPage.java
index d51913b..0de8968 100644
--- a/bundles/org.eclipse.equinox.p2.ui/src/org/eclipse/equinox/internal/p2/ui/dialogs/ProfileModificationWizardPage.java
+++ b/bundles/org.eclipse.equinox.p2.ui/src/org/eclipse/equinox/internal/p2/ui/dialogs/ProfileModificationWizardPage.java
@@ -156,7 +156,7 @@ public abstract class ProfileModificationWizardPage extends WizardPage {
 	}
 
 	public boolean performFinish() {
-		if (currentPlan != null && currentPlan.getStatus().isOK()) {
+		if (currentStatus != null && currentStatus.getSeverity() != IStatus.ERROR) {
 			ProfileModificationOperation op = createProfileModificationOperation(currentPlan);
 			ProvisioningOperationRunner.schedule(op, getShell(), StatusManager.SHOW | StatusManager.LOG);
 			return true;
